### PR TITLE
Add Feature Flag for Development of Progressive Onboarding

### LINKED ISSUE #5486: Create Feature Flag for Progressive Onboarding
### Description

We need to add a feature flag for progressive onboarding so that we can gate our work into PO done during the development phase until we are ready to launch.

### Acceptance criteria

* A feature flag is available for PO.
* We should also implement this into the dev tools.

### PR DESCRIPTION
**Pull request by dmallory42**
Fixes #5486 

#### Changes proposed in this Pull Request

- Adds a feature flag for progressive onboarding.
- Currently this is not used, but it will be used by future PRs working on the PO project. 

#### Testing instructions

* Check that the change makes sense and tests pass.
* Test the Dev Tools PR (https://github.com/Automattic/woocommerce-payments-dev-tools/pull/81) to ensure that the flag name matches and the flag can get set on/off correctly.

-------------------

- [x] Run `npm run changelog` to add a changelog file, choose `patch` to leave it empty if the change is not significant. You can add multiple changelog files in one PR by running this command a few times. 
- [x] Covered with tests (or have a good reason not to test in description ☝️)
- [x] Tested on mobile (or does not apply)

**Post merge**

- [x] Link to testing instructions from [release testing doc](https://github.com/Automattic/woocommerce-payments/wiki/Release-testing-instructions) following [these instructions](https://github.com/Automattic/woocommerce-payments/wiki/How-to-write-good-manual-testing-scenarios) : 'QA Testing Not Applicable
- [x] Add or update [critical flows](https://github.com/Automattic/woocommerce-payments/wiki/Critical-flows) and [testing instructions for critical flows](https://github.com/Automattic/woocommerce-payments/wiki/Testing-instructions-for-critical-flows), if applicable.
- [x] Add what's changed (description, screenshot, demo videos etc.) to the release announcement post, if applicable.

### CODE REVIEW

File: tests/unit/test-class-wc-payments-features.php [hunk 1]
```diff
@@ -160,6 +169,35 @@ function ( $pre_option, $option, $default ) {
 		$this->assertFalse( WC_Payments_Features::is_woopay_express_checkout_enabled() );
 	}
 
+	public function test_is_progressive_onboarding_enabled_returns_true() {
+		add_filter(
+			'pre_option__wcpay_feature_progressive_onboarding',
```
Comment (vladolaru) [inline tests/unit/test-class-wc-payments-features.php]: Could we use the `WC_Payments_Features::PROGRESSIVE_ONBOARDING_FLAG_NAME` constant to construct the filter name? Like:

```
'pre_option_' . WC_Payments_Features::PROGRESSIVE_ONBOARDING_FLAG_NAME,
```
I think this way tests will stay more in sync with whatever changes we might make to the constant's value.

**Comment by vladolaru** [comment]
Everything looks good on my side, @dmallory42! Good job.

Tests are green, and the WCPay Dev Tools with the changes in [the PO PR](https://github.com/Automattic/woocommerce-payments-dev-tools/pull/81) check out.

I left a comment about a small thing in the tests, but apart from that good to merge!

**Review by vladolaru — APPROVED** [review]
Everything looks good on my side, @dmallory42! Good job.

Tests are green, and the WCPay Dev Tools with the changes in [the PO PR](https://github.com/Automattic/woocommerce-payments-dev-tools/pull/81) check out.

I left a comment about a small thing in the tests, but apart from that good to merge!

File: tests/unit/test-class-wc-payments-features.php (hunk 1 above)
Comment (dmallory42) [reply tests/unit/test-class-wc-payments-features.php]: Good catch! I agree, updated in 7dea1f8d17335777b666a5e6378b65c2c99c1530.

**Comment by dmallory42** [comment]
Thanks a lot for the review, @vladolaru! 

Would you mind also leaving a review on the Dev Tools PR (since you tested it with this PR), please? I'm not sure if it's necessary to get an approving review before merging in that repo, but I'd prefer to err on the side of caution. 🙂

**Comment by vladolaru** [comment]
You are welcome, Dan! Doing the review on that other PR right away.

**Pull request merged**